src/apps/FlappyBird.jsx

```diff
@@ -0,0 +1,362 @@
+    });
+
+    // player
+    ctx.save();
+    ctx.translate(player.x, player.y);
+    ctx.rotate((player.v * Math.PI) / 18);
+    ctx.drawImage(sprite, -17, -10);
+    ctx.restore();
+
+    ctx.fillStyle = "#97FFFF";
+    ctx.fillRect(0, floor, wid, 2);
+    ctx.fillStyle = "#2F4F4F";
+    ctx.fillRect(0, floor + 1, wid, 1);
+    ctx.fillStyle = "#97FFFF";
+    ctx.fillRect(0, floor + 17, wid, 2);
+    ctx.fillStyle = "#2F4F4F";
+    ctx.fillRect(0, floor + 17, wid, 1);
+
+    // score
+    ctx.font = "30px monospace";
+    const hScore = "best:" + (score > high ? score : high);
+    const sWid = ctx.measureText(hScore).width;
+    const sY = 80;
+
+    ctx.fillStyle = "white";
+    ctx.fillText(score, 20, floor + sY);
+    ctx.fillText(hScore, wid - sWid - 20, floor + sY);
+  };
+
+  const adjust = () => {
+    if (trans % pillSpace === 0) {
+      let h;
+      pillars.push({
+        x: -trans + wid,
+        y: 0,
+        w: pillWid,
+        h: (h = Math.random() * (floor - 300) + 100),
+      });
+
+      pillars.push({
+        x: -trans + wid,
+        y: h + pillGap,
+        w: pillWid,
+        h: floor - h - pillGap,
+      });
+    }
+
+    pillars = pillars.filter((pill) => -trans < pill.x + pill.w);
+
+    player.v += gravity;
+    if (player.v > termVel) {
+      player.v = termVel;
+    }
+    player.y += player.v;
+
+    if (player.y < player.r) {
+      player.y = player.r;
+      player.v = 0;
+    }
+
+    for (const i in pillars) {
+      let pillIndex = i;
+      const pill = pillars[pillIndex];
+      if (
+        pill.x + trans < player.x + player.r &&
+        pill.x + pill.w + trans > player.x - player.r
+      ) {
+        if (
+          player.y - player.r > pill.y &&
+          player.y - player.r < pill.y + pill.h
+        ) {
+          colider = true;
+          running = false;
+          render();
+          break;
+        }
+        if (
+          player.y + player.r < pill.y + pill.h &&
+          player.y + player.r > pill.y
+        ) {
+          colider = true;
+          running = false;
+          render();
+          break;
+        }
+        if (!pill.passed && pillIndex % 2 == 1) {
+          score += 1;
+          pill.passed = true;
+        }
+      }
+    }
+
+    if (player.y + player.r - player.v > floor) {
+      player.y = floor - player.r;
+      running = false;
+      colider = true;
+      render();
+    }
+  };
+
+  document.onmousedown = () => {
+    if (running) {
+      player.v = thrust;
+    } else if (!colider) {
+      running = true;
+    } else {
+      if (score > high) {
+        localStorage.setItem("high", score);
+      }
+      init();
+    }
+  };
+  init();
+  function loop() {
+    if (running) {
+      adjust();
+    }
+    if (!colider) {
+      render();
+    }
+    requestAnimationFrame(loop);
+  }
+};
+
+function Renderer() {
+  useEffect(() => {
+    gameLogic();
+  }, []);
+
+  return (
+    <div>
+      <NyanComponentRender />
+    </div>
+  );
+}
```

Comment:
There isn't any need to have another wrapper over the `NyanComponentRender` component. You can simply add the same useEffect into the component and mention it's app object below.